esp/services/ws_workunits/ws_wudetails.cpp

```diff
@@ -0,0 +1,349 @@
+        wuStat.getScope(wuScope);
+        getScopeIdFromScopeName(wuId, wuScope);
+
+        if (limitToIdList && ids.bSearch(wuId.str(),funcCompareString)==NotFound)
+              continue;
+
+        if (maxDepth && getScopeDepth(wuScope) > maxDepth)
+            continue;
+
+        if (filterNestedScopeTypes && nestedScopeTypes.bSearch(wuStat.getScopeType(),funcCompareUnsignedShort)==NotFound)
+              continue;
+
+        if (filterByAttributeValue)
+        {
+            tmpForAttribSearch->setStatisticKind(wuStat.getKind());
+            aindex_t pos = arrayAttributeFilter.bSearch(tmpForAttribSearch,funcCompareAttribFilter);
+            if (pos!=NotFound)
+            {
+                CAttributeFilter & attributeFilter = arrayAttributeFilter.item(pos);
+                unsigned __int64 graphRawValue = wuStat.getValue();
+
+                if (attributeFilter.hasExactValue())
+                {
+                    if (graphRawValue != attributeFilter.getExactValue()) continue;
+                }
+                else
+                {
+                    if (attributeFilter.hasMinValue() && graphRawValue < attributeFilter.getMinValue()) continue;
+                    if (attributeFilter.hasMaxValue() && graphRawValue > attributeFilter.getMaxValue()) continue;
+                }
+            }
+        }
+        // Check if scope name has changed. If so, need to return the list attributes for previous scope
+        if (strcmp(currentScope.str(),wuScope.str())!=0)
+        {
+            if (currentScope.length()>0 && attribs.ordinality()>0)
+            {
+                Owned<IEspWUScope> respScope = buildRespWUScope(currentScope, currentScopeType, currentId, attribs);
+                respScopes.append(*respScope.getClear());
+                attribs.clear();
+            }
+            currentScope.set(wuScope.str());
+            currentScopeType.set(queryScopeTypeName(wuStat.getScopeType()));
+            currentId.set(wuId.str());
+        }
+
+        if (returnSpecifyReturnAttribs)
+        {
+            const unsigned sk = (unsigned) wuStat.getKind();
+            if (returnAttribList.bSearch(sk,funcCompareUnsignedShort)==NotFound)
+               continue;
+        }
+
+        if (wuStat.getTimestamp()<minVersion)
+            continue;
+        if (wuStat.getTimestamp()>maxTimestamp)
+            maxTimestamp = wuStat.getTimestamp();
+
+        Owned<IEspWUResponseAttribute> attrib = copyAttribs(wuStat, attribOptions);
+        attribs.append(*attrib.getClear());
+        ++statsAdded;
+    }
+    DBGLOG("Statistics processed %d, stats added %d", statsCount, statsAdded);
+    if (currentScope.length()>0 && attribs.ordinality()>0)
+    {
+        Owned<IEspWUScope> respScope = buildRespWUScope(currentScope, currentScopeType, currentId, attribs);
+        respScopes.append(*respScope.getClear());
+        attribs.clear();
+    }
+}
+
+IEspWUScope * CWUDetails::buildRespWUScope(SCMStringBuffer & scope,
```

Comment:
If you are passing constant strings into a function it is better to pass a const char * rather than a SCMStringBuffer & because it makes it more general.  It also isn't clear with a SCMStringBuffer & parameter whether it is an input or an output parameter.